clang/include/clang/Analysis/FlowSensitive/StorageLocation.h

```diff
@@ -88,12 +94,12 @@ class ScalarStorageLocation final : public StorageLocation {
 class RecordStorageLocation final : public StorageLocation {
 public:
   using FieldToLoc = llvm::DenseMap<const ValueDecl *, StorageLocation *>;
+  using SyntheticFieldMap = llvm::StringMap<StorageLocation *>;
```

Comment:
I was wondering, if we want this map to **not** be indexed by strings. An alternative would be to somehow access the `IdentifierTable` and create an identifier from the names and use the pointer to the identifier as the key. This has the potential to be a bit lighter on memory (identifiers only stored once as opposed to storing the keys in each RecordStorageLocation instance), and potentially a bit quicker lookup. 

That being said, I am not insisting on the change, and even if we want to do it, it is probably OK in a follow-up PR.

Reply:
This is an interesting idea, but I'm hesitant about mutating the `IdentifierTable` in something that is going to be used from a clang-tidy check, which will be sharing the `IdentifierTable` with other checks.

I suppose we could always create our own mapping from strings to integers or pointers of some sort, which could then serve as the keys for the `SyntheticFieldMap` -- but I would lean towards leaving this as a potential future optimization that we would do if this does indeed turn out to be a performance issue.